src/efi.rs

```diff
@@ -72,6 +72,17 @@ impl Efi {
         Ok(esp)
     }
 
+    fn esp_path_tmp(&self) -> Result<PathBuf> {
+        self.ensure_mounted_esp(Path::new("/"))
+            .map(|v| v.join(".EFI.tmp"))
+    }
```

Comment:
But...related to @travier's comment...it'd be good to try to scope this to the "lowest" directory we need to make the change.

For us that means e.g. we only affect `EFI/fedora` for example and not all of `EFI`.

Reply:
The `bootupd-state.json` also includes `BOOT/BOOTX64.EFI` and `BOOT/fbx64.efi` under `EFI/`, both are from shim-x64, we might need to skip the update?

Reply:
I think it should be reasonable to prepare both the new `BOOT` and `fedora` dirs and then do `fedora` first, `BOOT` second.

I don't think there is a dependency "across" those folders for the bootloader files, i.e. grub.efi may depend on an update shim.efi from the same folder but not on an update fbx.efy from the BOOT folder.

I'm not sure however.

Reply:
There were some elements in https://lists.fedoraproject.org/archives/list/devel@lists.fedoraproject.org/thread/VRJ7CRQUZ7SNEDW4AIOHKKJ7TMPTOX5A/

Reply:
https://github.com/rhboot/shim/pull/502

Reply:
> prepare both the new `BOOT` and `fedora` dirs and then do `fedora` first, `BOOT` second.

This is a little tricky that for the `apply_diff`, need to check if it is for `BOOT` or `fedora`, then how about using ".EFI.tmp"?

Reply:
See https://github.com/coreos/bootupd/pull/669#issuecomment-2182555634, will prepare both the new `BOOT` and `fedora` dirs and then do `fedora` first, `BOOT` second.

Reply:
> I don't think there is a dependency "across" those folders for the bootloader files, i.e. grub.efi may depend on an update shim.efi from the same folder but not on an update fbx.efy from the BOOT folder.

Hmmmm...this kind of rains on the parade here though given that `shim` includes files in these two directories, and we can't update it atomically without updating *everything* in `EFI`, which also includes potentially other files we don't manage.

Reply:
According to @travier 's suggestion, maybe can copy first sub dir as temp (like `BOOT/foo` -> `BOOT.tmp/foo`) and remember, then do remove/addition/change files in temp dir, finally scan temp and do exchange. I think it might be helpful when we do A/B updates. WDYT?

Reply:
Yes, this just means that shim updates won't be transactional, which is unfortunate as shim is the main one we need to update.

It should be an improvement...but...I'm a bit uncertain about saying it's enough that we unconditionally turn on auto-updates by default?

It feels like shim/grub update infrequently enough that in practice we could just accept the hit today of the non-transactionality and recommend turning on auto-updates in various distros/OSes?

I'd be roughly in favor of doing that in FCOS and fedora bootc today just to start.

Reply:
> It should be an improvement...but...I'm a bit uncertain about saying it's enough that we unconditionally turn on auto-updates by default?

Not sure, is it necessary to check the shim files integrity (like `sha512`) after sync to provide more confidence?